docs/best-practices.md

```diff
@@ -0,0 +1,26 @@
+# Best Practices
+
+## Guardian CDK builds in best practices:
```

Comment:
I'm not sure about this section. It's great to flag up that CDK aims to establish best practice(!) but this is potentially going to be quite a long and changing list. It might be better to document key points in the patterns and constructs themselves (which we could definitely do more of).

Reply:
Good point, I chatted about this with @akash1810 yesterday and he suggested instead of this section, having a link to the recommendations repo - will update this!